primer/test/Tests/Eval.hs

```diff
@@ -1459,7 +1437,10 @@ tasty_type_preservation =
 -- | Reductions do not interfere with each other
 -- if @i,j ∈ redexes e@  (and @i /= j@), and @e@ reduces to @e'@ via redex @i@
 -- then @j ∈ redexes e'@,
--- unless @j@ no longer exists in @e'@ or @j@ was a rename-binding which is no longer required
+-- unless one of the following is true
+-- - @j@ no longer exists in @e'@
+-- - @j@ was a rename-binding which is no longer required because @i@ was eliding a let
+-- - @i@ was a PushLet, which can intersperse itself inside the parts of @j@'s redex, blocking it
 tasty_redex_independent :: Property
 tasty_redex_independent =
```

Comment:
I do wonder if this test is getting a bit silly...

Reply:
Seeing as you've already written the code to fix it up, I see no good reason not to keep it around.

It's basically just the last case which makes it hard to follow, because of the explicit enumeration of the `i` cases. But then using a wildcard there would make the test less complete.